Realm/RLMRealmUtil.mm

```diff
@@ -36,11 +36,15 @@
 #import <unistd.h>
 
 // Global realm state
-static std::mutex s_realmCacheMutex;
+static dispatch_once_t s_realmCacheMutexDispatchToken;
+static std::mutex *s_realmCacheMutex;
```

Comment:
Ditto about this being `static std::mutex& s_realmCacheMutex = *new std::mutex;`.

Note that heap-allocating this mutex without doing similarly for the data it protects (`s_realmsPerPath`) won't be super helpful.